apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/service/state/impl/PollServiceImpl.java

```diff
@@ -0,0 +1,264 @@
+        final PollResultTable pollResultTable
+    ) {
+        this.blockChainInfoService = blockChainInfoService;
+        this.databaseManager = databaseManager;
+        this.pollTable = pollTable;
+        this.pollResultTable = pollResultTable;
+        this.converter = new IteratorToStreamConverter<>();
+    }
+
+    @Override
+    public void checkPolls(int currentHeight) {
+        // select all Polls where 'finish_height' is EQUAL (DbClause.Op.EQ) then specified height value
+        try (DbIterator<Poll> polls = pollTable.getPollsFinishingAtHeight(currentHeight)) {
+            int index = 0;
+            for (Poll poll : polls) {
+                try {
+                    List<PollOptionResult> results = countResults(
+                        poll.getVoteWeighting(),
+                        currentHeight,
+                        poll.getId(),
+                        poll.getAccountId(),
+                        poll.getOptions().length
+                    );
+                    log.trace("checkPolls: height = {}, [{}] PollId = {} has = {}", currentHeight, index, poll.getId(), results.size());
+                    pollResultTable.insert(results);
+                    log.trace("checkPolls: height = {}, [{}] PollId = {} checked : {}", currentHeight, index, poll.getId(), results);
+                    index++;
+                } catch (RuntimeException e) {
+                    log.error("Couldn't count RollResult for poll {} at height = {}", poll.getId(), currentHeight, e);
+                }
+            }
+        }
+    }
+
+    @Override
+    public Poll getPoll(long id) {
+        return pollTable.getPoll(id);
+    }
+
+    @Override
+    public Stream<Poll> getPollsFinishingAtOrBefore(int height, int from, int to) {
+        return converter.convert(pollTable.getPollsFinishingAtOrBefore(height, from, to));
+    }
+
+    @Override
+    public DbIterator<Poll> getPollsFinishingBelowHeight(int height, int from, int to) {
+        return pollTable.getPollsFinishingBelowHeight(height, from, to);
+    }
+
+    @Override
+    public Stream<Poll> getAllPolls(int from, int to) {
+        return converter.convert(pollTable.getAll(from, to));
+    }
+
+    @Override
+    public Stream<Poll> getActivePolls(int from, int to) {
+        return converter.convert(pollTable.getActivePolls(from, to, blockChainInfoService.getHeight()));
+    }
+
+    @Override
+    public Stream<Poll> getPollsByAccount(long accountId, boolean includeFinished, boolean finishedOnly, int from, int to) {
+        return converter.convert(
+            pollTable.getPollsByAccount(accountId, includeFinished, finishedOnly, from, to, blockChainInfoService.getHeight())
+        );
+    }
+
+    @Override
+    public Stream<Poll> getVotedPollsByAccount(long accountId, int from, int to) throws AplException.NotValidException {
+        return converter.convert(
+            pollTable.getVotedPollsByAccount(databaseManager.getDataSource(), accountId, from, to)
```

Comment:
We can remove using DataSource as input parameter for method. See suggestions above for 'PollTable' component.

Reply:
done